utils/scarb-test-support/src/proc_macro_server.rs

```diff
@@ -0,0 +1,166 @@
+use crate::command::Scarb;
+use scarb_proc_macro_server_types::jsonrpc::RequestId;
+use scarb_proc_macro_server_types::jsonrpc::ResponseError;
+use scarb_proc_macro_server_types::jsonrpc::RpcRequest;
+use scarb_proc_macro_server_types::jsonrpc::RpcResponse;
+use scarb_proc_macro_server_types::methods::Method;
+use std::collections::HashMap;
+use std::io::BufRead;
+use std::io::BufReader;
+use std::io::Lines;
+use std::io::Write;
+use std::marker::PhantomData;
+use std::path::Path;
+use std::process::Child;
+use std::process::ChildStdin;
+use std::process::ChildStdout;
+use std::process::Stdio;
+
+pub const SIMPLE_MACROS: &str = r#"
+use cairo_lang_macro::{
+    ProcMacroResult,
+    TokenStream,
+    attribute_macro,
+    inline_macro,
+    derive_macro,
+    executable_attribute
+};
+
+executable_attribute!("some_executable");
+
+#[attribute_macro]
+pub fn some(_attr: TokenStream, token_stream: TokenStream) -> ProcMacroResult {
+    ProcMacroResult::new(token_stream)
+}
+
+#[inline_macro]
+pub fn inline_some(token_stream: TokenStream) -> ProcMacroResult {
+    ProcMacroResult::new(token_stream)
+}
+
+#[derive_macro]
+fn some_derive(_token_stream: TokenStream)-> ProcMacroResult {
+    ProcMacroResult::new(TokenStream::new("impl SomeImpl of SomeTrait {}".to_string()))
+}
+"#;
+
+pub struct PendingRequest<M: Method> {
+    id: RequestId,
+    _data: PhantomData<M>,
+}
+
+impl<M: Method> PendingRequest<M> {
+    fn new(id: RequestId) -> Self {
+        Self {
+            id,
+            _data: Default::default(),
+        }
+    }
+}
+
+pub struct Response<M: Method> {
```

Comment:
>Cannot we just always return Result<<M as Method>::Response, ResponseError> or RpcResponse? This seems unnecessary

This will not work if we want to do something more complex than 1 by 1 execution

Reply:
My bad, had `PendingRequest` in mind, you are right